tools/lsp/preview/ui/property_view.rs

```diff
@@ -178,19 +180,166 @@ fn convert_number_literal(
+                name: if palette_name.is_empty() {
+                    None
+                } else {
+                    Some(palette_name.iter().rev().join("."))
+                },
+            };
+        }
+        expression_tree::Expression::PropertyReference(source)
+            if source.element().borrow().base_type == langtype::ElementType::Global =>
+        {
+            println!("Following property reference");
+            let elem = source.element();
+            let elem = elem.borrow();
+            if let Some(component_name) = elem.enclosing_component.upgrade().map(|c| c.id.clone()) {
+                if let Some(binding) = elem.bindings.get(source.name()) {
+                    palette_name.push(source.name().to_string());
+                    palette_name.push(component_name.to_string());
+
+                    let struct_level = palette_name.len().saturating_sub(3);
+
+                    return extract_color_from_evaluated_expression_impl(
+                        &binding.borrow().expression,
+                        palette_name,
+                        struct_level,
+                    );
+                }
+            }
+        }
+        expression_tree::Expression::StructFieldAccess { base, name } => {
+            println!("Stripping StructFieldAccess");
+            palette_name.push(name.to_string());
+            return extract_color_from_evaluated_expression_impl(base, palette_name, struct_level);
+        }
+        expression_tree::Expression::Struct { values, .. } => {
+            println!("Stripping Struct");
+            if let Some(part) = palette_name.get(struct_level) {
+                eprintln!("  => field: {part}");
+                if let Some(value) = values.get(&SmolStr::from(part)) {
+                    eprintln!("    => with expr: {value:?}");
+                    return extract_color_from_evaluated_expression_impl(
+                        value,
+                        palette_name,
+                        struct_level.saturating_sub(1),
+                    );
+                }
+            }
+        }
+        expression_tree::Expression::Condition { true_expr, .. } => {
+            println!("Stripping Condition");
+            return extract_color_from_evaluated_expression_impl(
+                true_expr,
+                palette_name,
+                struct_level,
+            );
+        }
+        expression_tree::Expression::CodeBlock(cb) => {
+            println!("Stripping CodeBlock");
+            let Some(expr) = cb.first() else {
+                return ExtractedColor::default();
+            };
+            return extract_color_from_evaluated_expression_impl(expr, palette_name, struct_level);
+        }
+        expression_tree::Expression::StoreLocalVariable { value, .. } => {
+            println!("Stripping StoreLocalVariable");
+            return extract_color_from_evaluated_expression_impl(value, palette_name, struct_level);
+        }
+        _ => {
```

Comment:
You could also support Expression::LinearGradient and Expression::RadialGradient

Then you can use the same function for both color and gradient and don't need special code for the gradients.

Reply:
Yes, I probably need to extend that to brushes eventually, but let's wait and see what our designers say about that though.